doc/quickstart.md

```diff
@@ -101,6 +101,93 @@ the route to the testserver needs to be added from another python console.
 >>> plc.close()
 ```
 
+### ADS Symbol
+
+#### Symbol creation
+
+Reading from or writing to an ADS variable (= an ADS symbol) can be done through an `AdsSybmol` instance:
+
+```python
+>>> import pyads
+>>> plc = pyads.Connection('127.0.0.1.1.1', pyads.PORT_SPS1)
+>>> plc.open()
+>>> symbol = plc.get_symbol('global.bool_value')
+```
+
+The address and type of the symbol will be automatically determined using a READ_WRITE request to the ADS server, based on the variable name. This lookup is skipped when all the information has already been provided:
+
+```python
+>>> import pyads
+>>> plc = pyads.Connection('127.0.0.1.1.1', pyads.PORT_SPS1)
+>>> plc.open()
+# Identical to `plc.get_symbol('global.bool_value')`, remaining info will be looked up:
+>>> symbol = pyads.AdsSymbol(plc, 'global.bool_value')
```

Comment:
I wonder what the exact difference is between plc.get_symbol and pyads.AdsSymbol(plc, ...). They seem to do two methods for the same task at least from a user-point-of-view I think. Could we just expand plc.get_symbol to also support passing  all informations and encourage plc.get_symbol?

Reply:
> Just wondering if we could introduce a device_notification parameter on symbol creation that (if set to true) automatically creates a device notification which changes the buffered value as soon as the plc variable is changed. This way value is always the actual value from the plc.

If only the buffer changes without triggering anything else, what is the point? With such a default callback, `symbol.value` will again be almost identical to `symbol.read()`. If a user wants the most recent value, they can call `.read()` directly.

Reply:
> I wonder what the exact difference is between plc.get_symbol and pyads.AdsSymbol(plc, ...). They seem to do two methods for the same task at least from a user-point-of-view I think. Could we just expand plc.get_symbol to also support passing all informations and encourage plc.get_symbol?

There is not real difference now. We could advocate the use of either the factory method or direct object initialization, and not the other. Does either have your preference?

Reply:
> If only the buffer changes without triggering anything else, what is the point? With such a default callback, `symbol.value` will again be almost identical to `symbol.read()`. If a user wants the most recent value, they can call `.read()` directly.

I think of a use-case like a plot where the client plots plc values periodically like every 100ms or so. You basically read the current value from value and add it to the plot. With the internal device notification you don't need to poll from the plc every cycle. Instead only changed values will be pushed to the buffer and can be read much faster into the graph.

Reply:
> There is not real difference now. We could advocate the use of either the factory method or direct object initialization, and not the other. Does either have your preference?

I prefer the factory function as this is more intuitive for me because we use "plc." for everything anyway. But I leave this decisision up to you. If you like to use object initializiation that's fine. But as you say we should just advocate one way of doing it in the Quickstart.

Reply:
I updated the `plc.get_symbol()` method to match with the `AdsSymbol` constructor. I also removed the 'manual' object creation, so only the factory method is used.

I see your point about auto-updating the buffer, so I added such a feature.
I felt like the constructor already had enough arguments, so I've added a method to enable auto-update. Thoughts?

Reply:
This looks very nice. The naming `set_auto_update` is fitting I think. Anyway I think we could add one more parameter `auto_update=False` to `AdsSymbol.__init__`. By default it will be false but if someone likes to enable it it would be just a breathe to add `auto_update=True`. Especially if one creates multiple symbols I think it would be very convenient to have this handy parameter on initialization instead of writing an extra line for each symbol.

Reply:
Alright, that can't hurt. I've added it.